framework/src/userobjects/TimedSubdomainModifier.C

```diff
@@ -0,0 +1,298 @@
+    const auto it = find(_names.begin(), _names.end(), s);
+    if (it == _names.end())
+      mooseError("Could not find '", s, "' in header of file ", _file_name, ".");
+    _blocks_from_column = std::distance(_names.begin(), it);
+  }
+  else if (isParamValid("blocks_from_column_index"))
+  {
+    _blocks_from_column = getParam<size_t>("blocks_from_column_index");
+  };
+
+  size_t _blocks_to_column = 2;
+  if (isParamValid("blocks_to_column_text"))
+  {
+    const auto s = getParam<std::string>("blocks_to_column_text");
+    const std::vector<std::string> _names = file.getNames();
+    const auto it = find(_names.begin(), _names.end(), s);
+    if (it == _names.end())
+      mooseError("Could not find '", s, "' in header of file ", _file_name, ".");
+    _blocks_to_column = std::distance(_names.begin(), it);
+  }
+  else if (isParamValid("blocks_to_column_index"))
+  {
+    _blocks_to_column = getParam<size_t>("blocks_to_column_index");
+  };
+
+  size_t max_needed_column_index = std::max({_time_column, _blocks_from_column, _blocks_to_column});
+
+  std::vector<std::vector<std::string>> data = file.getData();
+
+  if (data.size() < max_needed_column_index)
+    mooseError("data must contain at least " + std::to_string(max_needed_column_index) +
+                   " columns in file ",
+               _file_name);
+
+  std::vector<std::string> strTimes = data[_time_column];
+  std::vector<std::string> strBlockFrom = data[_blocks_from_column];
+  std::vector<std::string> strBlockTo = data[_blocks_to_column];
+
+  size_t n_rows = strTimes.size();
+
+  // some sanity checks
+  if (n_rows == 0)
+    mooseError("empty sequence in file ", _file_name);
+  if (n_rows != strBlockFrom.size())
+    mooseError("inconsistent data in ", _file_name);
+  if (n_rows != strBlockTo.size())
+    mooseError("inconsistent data in ", _file_name);
+
+  // resize variables to fit the data
+  _times.resize(n_rows);
+  _blocks_from.resize(n_rows);
+  _blocks_to.resize(n_rows);
+
+  // parse the data given by the user
+  const std::shared_ptr<MooseMesh> _mesh = _app.actionWarehouse().mesh();
+  std::transform(
+      strTimes.begin(), strTimes.end(), _times.begin(), [](std::string x) { return std::stod(x); });
+  std::transform(strBlockFrom.begin(),
+                 strBlockFrom.end(),
+                 _blocks_from.begin(),
+                 [_mesh](std::string x) { return _mesh->getSubdomainID(x); });
+  std::transform(strBlockTo.begin(),
+                 strBlockTo.end(),
+                 _blocks_to.begin(),
+                 [_mesh](std::string x) { return _mesh->getSubdomainID(x); });
+}
+
+void
+TimedSubdomainModifier::initialize()
+{
+  TimedElementSubdomainModifier::initialize();
+}
+
+std::vector<real>
+TimedSubdomainModifier::onGetTimes()
+{
+  // just return our local array of times
+  return _times;
+}
+
+SubdomainID
+TimedSubdomainModifier::onComputeSubdomainID(real t_from_exclusive, real t_to_inclusive)
```

Comment:
```suggestion
TimedSubdomainModifier::onComputeSubdomainID(const Real t_from_exclusive, const Real t_to_inclusive)
```
surprised we even define real